### ISSUE TITLE
Dorado 0.3.4 poor performance basecalling SQK-RNA002 

### ISSUE THREAD

**Issue by pre-mRNA** [nanoporetech/dorado#383]
Hi, 

To test Dorado vs Guppy for direct RNA basecalling, I am basecalling ~1.3M direct RNA reads using Dorado 0.3.4 on a cluster with 4 x Tesla Volta V100 32GB GPUs. 

Strangely, for a library with ~300k reads, the basecalling finishes in < 30m, but with a library of 1.3M reads, it takes > 12 hours.

To optimise performance, I've tried to basecall with either 2 or 4 V100 GPUs,I've tried to basecall while the SQK-RNA002 FAST5 files are stored either on a lustre filesystem or directly on a local SSD, and I've set the --device flag to either 'default' or 'cuda:0,1' when basecalling with 2 GPUs. 

In any of the cases, the RNA basecalling performance is very weak. While I can basecall larger DNA FAST5 sets in <1 hour, the RNA basecalling for >1M reads takes > 12 hours of walltime, which is unexpected. 

My current command (with 2 GPUs) is: 

dorado basecaller "${model}" "./fast5_dir/" -v -b 0 -r --device cuda:0,1 --emit-fastq

And this takes more than 12 hours to basecall ~130gb of SQK-RNA002 FAST5 files from local disk using 2 Volta V100s.

Any ideas what I can do to improve performance?

**Comment by pre-mRNA** [comment]
Strangely, the gpu utilisation seems to be high, at least just now: 

+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 535.54.03              Driver Version: 535.54.03    CUDA Version: 12.2     |
|-----------------------------------------+----------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |         Memory-Usage | GPU-Util  Compute M. |
|                                         |                      |               MIG M. |
|=========================================+======================+======================|
|   0  Tesla V100-SXM2-32GB           On  | 00000000:3E:00.0 Off |                    0 |
| N/A   43C    P0             287W / 300W |  13071MiB / 32768MiB |    100%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
|   1  Tesla V100-SXM2-32GB           On  | 00000000:B1:00.0 Off |                    0 |
| N/A   44C    P0             333W / 300W |  13071MiB / 32768MiB |    100%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
                                                                                         
+---------------------------------------------------------------------------------------+
| Processes:                                                                            |
|  GPU   GI   CI        PID   Type   Process name                            GPU Memory |
|        ID   ID                                                             Usage      |
|=======================================================================================|
|    0   N/A  N/A    670149      C   dorado                                    13068MiB |
|    1   N/A  N/A    670149      C   dorado                                    13068MiB |
+---------------------------------------------------------------------------------------+

Yet, Dorado has basecalled only ~750k reads in 16 hours

**Comment by sklages** [comment]
No explanation for the issue but [nvtop](https://github.com/Syllo/nvtop) is a useful tool for inspecting GPU load over a time range.

**Comment by tijyojwad** [comment]
`dorado` is optimized for data loading from `pod5`. Can you convert your fast5 to pod5 format and try the same experiment (both from lustre filesystem and local SSD)?

**Comment by pre-mRNA** [comment]
Sure. I ran a quick test comparing POD5 to FAST5 for Dorado 0.3.4 basecalling.

I used a test-set of 1.55M SQK-RNA002 reads and basecalled them from local disk using 2 x V100 32GB GPUs. I performed basecalling twice, once with the original multi-FAST5s, and then with the same data converted to POD5 (using command 'pod5 convert fast5 "${fast5_dir}" --output "${pod5_path}" -t 104 --one-to-one ./

1. Basecalling from FAST5 stored on local disk 

Basecalling from FAST5 input, I only managed to call 1.05M reads before the walltime limit of 24h was reached, giving a performance of around 44k reads per hour. 

2. Basecalling from POD5 stored on local disk 

With the same data stored in POD5 format, I managed to basecall everything in 40 minutes of walltie, giving a performance of around 2.3M reads called per hour. 

The conclusion here is to convert FAST5 to POD5 files before basecalling SQK-RNA002 FAST5 files. Dorado seems to run ~50x faster this way. 

Thanks for your help!
